pyfixest/estimation/feglm_.py

```diff
@@ -0,0 +1,520 @@
+            self._X.to_numpy(),
+        )
+        if self._fe is not None:
+            self._fe = self._fe.to_numpy()
+            if self._fe.ndim == 1:
+                self._fe = self._fe.reshape((self._N, 1))
+
+    def get_fit(self):
+        "Fit the GLM model via iterated weighted least squares."
+        _Y = self._Y
+        _X = self._X
+        _fe = self._fe
+        _N = self._N
+        _convergence = self.convergence  # False
+        _maxiter = self.maxiter
+        _tol = self.tol
+        _fixef_tol = self._fixef_tol
+        _solver = self._solver
+
+        # initialize
+
+        beta = np.zeros(_X.shape[1])
+        eta = np.zeros(_N)
+        mu = self._get_mu(theta=eta)
+        deviance = self._get_deviance(_Y.flatten(), mu)
+        deviance_old = deviance.copy() + 1
+
+        for r in range(_maxiter):
+            if r == 0:
+                pass
+            else:
+                converged = self._check_convergence(
+                    crit=self._get_diff(deviance=deviance, last=deviance_old),
+                    tol=_tol,
+                    r=r,
+                    maxiter=_maxiter,
+                    model=self._method,
+                )
+                if converged:
+                    break
+
+            # Step 1: _get weights w_tilde(r-1) and v(r-1) (eq. 2.5)
+            detadmu = self._update_detadmu(mu=mu)
+            # v = self._update_v(y=_Y.flatten(), mu=mu, detadmu=detadmu)
+            W = self._update_W(mu=mu)
+
+            # Step 2: _get v_tilde(r-1) and X_tilde(r-1) (eq. 3.2)
+            W_tilde = self._update_W_tilde(W=W)
+            X_tilde = self._update_X_tilde(W_tilde=W_tilde, X=_X)
+            v_tilde = self._update_v_tilde(
+                y=_Y.flatten(), mu=mu, W_tilde=W_tilde.flatten(), detadmu=detadmu
+            )
+
+            # Step 3 compute v_dotdot(r-1) and X_dotdot(r-1) - demeaning
+            v_dotdot, X_dotdot = self.residualize(
+                v=v_tilde,
+                X=X_tilde,
+                flist=_fe,
+                weights=W_tilde.flatten(),
+                tol=_fixef_tol,
+            )
+
+            # Step 4: compute (beta(r) - beta(r-1)) and check for convergence, _update beta(r-1) s(eq. 3.5)
+            beta_update_diff = self._update_beta_diff(
+                X_dotdot=X_dotdot, v_dotdot=v_dotdot
+            )
+
+            # Step 5: _update using step halfing (if required)
+
+            deviance_old = deviance.copy()
+
+            beta, eta, mu, deviance = self._update_eta_step_halfing(
+                Y=_Y,
+                beta=beta,
+                eta=eta,
+                mu=mu,
+                deviance=deviance_old,
+                beta_update_diff=beta_update_diff,
+                W_tilde=W_tilde.flatten(),
+                v_tilde=v_tilde,
+                v_dotdot=v_dotdot,
+                X_dotdot=X_dotdot,
+                deviance_old=deviance_old,
+                step_halfing_tolerance=1e-12,
```

Comment:
do you a case where the user would need to modify the `step_halfing_tolerance` parameter?

Reply:
I checked the `glmnet` API, and they do not seem to support setting the step halfing tolerance: [link](https://glmnet.stanford.edu/reference/glmnet.html). For now I would maybe not support it, but add it in case users request it?